internal/autodetect/operator.go

```diff
@@ -0,0 +1,37 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package autodetect
+
+import (
+	"fmt"
+	"os"
+)
+
+func GetOperatorNamespace() (string, error) {
+	nsBytes, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		return "", err
+	}
+	return string(nsBytes), nil
+}
+
+func GetOperatorServiceAccount() (string, error) {
+	saEnvVar := "SERVICE_ACCOUNT_NAME"
+	sa := os.Getenv(saEnvVar)
+	if sa == "" {
+		return sa, fmt.Errorf("%s env variable not found", saEnvVar)
```

Comment:
So if this doesn't exist, we just log the error and continue as if we didn't have the permissions? That sounds reasonable to me.

Reply:
Yes